### ISSUE TITLE
enable extended resources in kubemark

### ISSUE THREAD

**Issue by xial-thu** [kubernetes/kubernetes#86378]
**What would you like to be added**:
Enable extended resources in kubemark by exposing flags such as `enable-extended-resource` or `extended-resources-conf-path`

**Why is this needed**:
Kubernetes is more and more popular in AI. Due to GPUs are not able to be shared among pods, the scenario of scheduling AI jobs is quite different from common cases. And extended scheduler such as kube-batch or volcano are designed for AI/ML scenario. The performance of those schedulers should be tested and whether there're unacceptable resource race conditions if using multiple schedulers at the same time should be evaluated as well. So I'm working on doing some benchmark on kubemark. 

But kubemark by default sets scalar resource to nil and does not expose interfaces for it.

**How this can be solved**
I've done a simple hack. In `GetDevicePluginResourceCapacity()` I return something else.
``` golang
func (cm *containerManagerStub) GetDevicePluginResourceCapacity() (v1.ResourceList, v1.ResourceList, []string) {
	return cm.extendedResource, cm.extendedResource, []string{}
}
```

by adding a resourceList in `containerManagerStub`

``` golang
type containerManagerStub struct {
	shouldResetExtendedResourceCapacity bool
	extendedResource                    v1.ResourceList
}
```

Also, I'll suggest exposing two more flags
* `enable-extended-resources`: bool
* `extended-resources-conf-path`: configuration path, whose content is YAML formatted

**Comment by hzliangbin** [comment]
@xial-thu any progress

**Comment by xial-thu** [comment]
> @xial-thu any progress

Sorry, I lost the source code...

Several years ago I hacked k8s 1.15 and implemented this feature. And I integrate it into a IaC tool in my department. But I did not contribute to the community due to it's only a dirty hack.

Review the issue now, I would suggest adding a provider to virtual-kubelet, which is reletively more active and modular. Actually I've been focusing on AI-infra for LLM training, not just kubernetes in recently years, so I'm afraid there're not any progress.
